src/sentry/replays/query.py

```diff
@@ -119,6 +120,10 @@ def query_replay_viewed_by_ids(
     else:
         project_ids = [project_id]
 
+    for project_id in project_ids:
+        if project_id in options.get("replay.viewed-by.project-denylist"):
+            return []
```

Comment:
This will cause a 404.  https://github.com/getsentry/sentry/blob/997cf1157084c4186b7714c0a51f6a2dfc86b969/src/sentry/replays/endpoints/project_replay_viewed_by.py#L71-L81

We should return something contextual.  Maybe an error message.